test/producers/application_producer_test.rb

```diff
@@ -3,34 +3,36 @@
 require 'test_helper'
 
 class ApplicationProducerTest < ActiveSupport::TestCase
-  class MockProducer < ApplicationProducer
-    BROKERS = ['broker1'].freeze
+  setup do
+    @_kafka_settings = Settings.kafka
+  end
+
+  teardown do
+    Settings.kafka = @_kafka_settings
```

Comment:
Does `Settings.reload!` not do it?

Reply:
For some reason it did not. I can retry.

Reply:
It worked at the end. Thanks.